### DISCUSSION TITLE
@storybook/nextjs: add support for unstable_getImgProps

### DISCUSSION THREAD

**Discussion by tremby** [storybookjs/storybook#23588] [Ideas]
### Is your feature request related to a problem? Please describe.

I'm using [`unstable_getImgProps`](https://github.com/vercel/next.js/discussions/25393#discussioncomment-6310618) so I can use `next/image`'s features with the `<picture>`/`<source>` elements, as explained in the linked comment.

When I try to use this in a story, I currently get an error: `next_image__WEBPACK_IMPORTED_MODULE_2__.unstable_getImgProps is not a function`.

The same code works fine in my Next application, so that leads me to believe it's a missing mock in `@storybook/nextjs`.

### Describe the solution you'd like

A working Storybook-friendly implementation of `unstable_getImgProps`.

### Are you able to assist to bring the feature to reality?

no

### Additional context

I'm aware it's unstable at present so I understand if an implementation won't be added until the upstream feature becomes stable.

**Comment by timheerwagen** [comment]
I would also be very interested in this feature.

**Comment by ygkn** [comment]
From Next.js 14.1, `unstable_getImgProps` is now stable (with renamed as `getImageProps`).

https://nextjs.org/blog/next-14-1#nextimage-support-for-picture-and-art-direction

**Reply by valentinpalkovic** [reply]
Since it is stable now, we can consider supporting it in Storybook. Is anyone interested in implementing this functionality? I can definitely support if needed!

**Reply by ygkn** [reply]
I am interested in implementing this feature. I have created a pull request for it. #25747
